internal/client/config.go

```diff
@@ -232,14 +249,20 @@ func NewFromConfigFile(filename string) (*client.ClientWithResponses, error) {
 	if err := config.Validate(); err != nil {
 		return nil, err
 	}
-	if err := config.Flatten(); err != nil {
+	return config, config.Flatten()
+}
+
+// NewFromConfigFile returns a new FlightCtl API client using the config read from the given file.
+func NewFromConfigFile(filename string) (*client.ClientWithResponses, error) {
+	config, err := ParseConfigFile(filename)
+	if err != nil {
 		return nil, err
 	}
 	return NewFromConfig(config)
 }
 
 // WriteConfig writes a client config file using the given parameters.
-func WriteConfig(filename string, server string, tlsServerName string, ca *crypto.TLSCertificateConfig, client *crypto.TLSCertificateConfig) error {
+func WriteConfig(filename string, cfg config.Config, tlsServerName string, ca *crypto.TLSCertificateConfig, client *crypto.TLSCertificateConfig) error {
```

Comment:
Ok, i've changed this back so that the server config is not required. Now with the other changes in the code, we are passing if auth is disabled or not.